lib/propolis-client/src/instance_spec/devices.rs

```diff
@@ -257,26 +282,46 @@ impl DeviceSpec {
     pub fn is_migration_compatible(
         &self,
         other: &Self,
-    ) -> Result<(), (String, SpecMismatchDetails)> {
-        self.board
-            .is_migration_compatible(&other.board)
-            .map_err(|e| ("board".to_string(), e))?;
+    ) -> Result<(), MigrationCompatibilityError> {
```

Comment:
It's a little unfortunate that we have 3 methods with this name with `&Self, &Self` as the args but differ in the return type. Would it be worth slightly renaming them? Or at the least I think the top-level ones could have slightly different ones?

Reply:
59ca882 implements some more descriptive names. I was hesitant to do this at first because having an `impl FooBarBaz` with routine names that included `foo_bar_baz` seemed a little strange, but I rather like how `Preamble::is_migration_compatible` looks now. Thanks for the nudge on this one!